crypto/rand/drbg_lib.c

```diff
@@ -417,6 +417,39 @@ int RAND_DRBG_generate(RAND_DRBG *drbg, unsigned char *out, size_t outlen,
     return 1;
 }
 
+/*
+ * Generates |outlen| random bytes and stores them in |out|. It will
+ * using the given |drbg| to generate the bytes.
+ *
+ * If |drbg| is NULL it assumes RAND_get_rand_method() != RAND_OpenSSL()
+ * and uses RAND_bytes() instead.
+ *
+ * Requires that drbg->lock is already locked for write, if non-null.
+ *
+ * Returns 1 on success 0 on failure.
+ */
+int RAND_DRBG_bytes(RAND_DRBG *drbg, unsigned char *out, size_t outlen)
+{
+    unsigned char *additional = NULL;
+    size_t additional_len;
+    size_t ret;
+
+    if (drbg == NULL) {
+        if (outlen > INT_MAX)
+            return 0;
+        return RAND_bytes(out, (int)outlen);
```

Comment:
I don't like this automagical fallback to RAND_bytes(). The default RAND_METHOD  (RAND_OpenSSL()) is implemented on top of the RAND_DRBG api, so RAND_DRBG's implementation should not refer back to  RAND_METHODs.

Also, no other RAND_DRBG_xxx() api call accepts a NULL pointer as drbg. It's the caller's duty to choose properly between RAND_bytes() and RAND_DRBG_bytes(), like in `ssl_randbytes`.